networkx/generators/tests/test_atlas.py

```diff
@@ -1,62 +1,75 @@
-        import platform
-        if platform.python_implementation()=='Jython':
-            raise SkipTest('graph atlas not available under Jython.')
-        import networkx.generators.atlas as atlas
+from itertools import groupby
+
+from nose.tools import assert_equal
+from nose.tools import assert_less_equal
+from nose.tools import raises
+
+import networkx as nx
+from networkx import graph_atlas
+from networkx import graph_atlas_g
+from networkx.generators.atlas import NUM_GRAPHS
+from networkx.utils import pairwise
+
+
+class TestAtlasGraph(object):
+    """Unit tests for the :func:`~networkx.graph_atlas` function."""
+
+    @raises(ValueError)
+    def test_index_too_small(self):
+        graph_atlas(-1)
+
+    @raises(ValueError)
+    def test_index_too_large(self):
+        graph_atlas(NUM_GRAPHS)
+
+    def test_graph(self):
+        G = graph_atlas(6)
+        assert_equal(set(G), set(range(3)))
+        assert_equal(list(G.edges()), [(0, 1), (0, 2)])
+
+
+class TestAtlasGraphG(object):
+    """Unit tests for the :func:`~networkx.graph_atlas_g` function."""
 
     def setUp(self):
-        self.GAG=atlas.graph_atlas_g()
+        self.GAG = graph_atlas_g()
 
     def test_sizes(self):
-        G=self.GAG[0]
-        assert_equal(G.number_of_nodes(),0)
-        assert_equal(G.number_of_edges(),0)
+        G = self.GAG[0]
+        assert_equal(G.number_of_nodes(), 0)
+        assert_equal(G.number_of_edges(), 0)
 
-        G=self.GAG[7]
-        assert_equal(G.number_of_nodes(),3)
-        assert_equal(G.number_of_edges(),3)
+        G = self.GAG[7]
+        assert_equal(G.number_of_nodes(), 3)
+        assert_equal(G.number_of_edges(), 3)
 
     def test_names(self):
-        i=0
-        for g in self.GAG:
-            name=g.name
-            assert_equal(int(name[1:]),i)
-            i+=1
-
-    def test_monotone_nodes(self):
-        # check for monotone increasing number of nodes
-        previous=self.GAG[0]
-        for g in self.GAG:
-            assert_false(len(g)-len(previous) > 1)
-            previous=g.copy()
-
-    def test_monotone_nodes(self):
-        # check for monotone increasing number of edges
-        # (for fixed number of nodes)
-        previous=self.GAG[0]
-        for g in self.GAG:
-            if len(g)==len(previous):             
-                assert_false(g.size()-previous.size() > 1)
-            previous=g.copy()
-
-    def test_monotone_degree_sequence(self):
-        # check for monotone increasing degree sequence
-        # (for fixed number f nodes and edges)
-        # note that 111223 < 112222
-        previous=self.GAG[0]
-        for g in self.GAG:
-            if len(g)==0: 
-                continue
-            if len(g)==len(previous) & g.size()==previous.size():
-                deg_seq=sorted(g.degree().values())
-                previous_deg_seq=sorted(previous.degree().values())
-                assert_true(previous_deg_seq < deg_seq)
-                previous=g.copy()
```

Comment:
In the old test, this line should not have been indented here, it should have been at the same level as the `if ...` above. This caused the test to pass trivially, as it wasn't checking any of the assertions. Dedenting this line reveals the same test failure as my new test below.